qiskit/algorithms/gradients/base_qgt.py

```diff
@@ -0,0 +1,342 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Abstract base class of the Quantum Fisher Information (QFI).
+"""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from collections.abc import Sequence
+from copy import copy
+
+import numpy as np
+
+from qiskit.algorithms import AlgorithmJob
+from qiskit.circuit import Parameter, ParameterExpression, QuantumCircuit
+from qiskit.primitives import BaseEstimator
+from qiskit.primitives.utils import _circuit_key
+from qiskit.providers import Options
+from qiskit.transpiler.passes import TranslateParameterizedGates
+
+from .qgt_result import QGTResult
+from .utils import (DerivativeType, GradientCircuit, _assign_unique_parameters,
+                    _make_gradient_parameter_set,
+                    _make_gradient_parameter_values)
+
+
+class BaseQGT(ABC):
+    r"""Base class to computes the Quantum Geometric Tensor(QGT) given a pure,
+    parameterized quantum state. QGT is defined as:
+
+    .. math::
+
+        \mathrm{QGT}_{kl}= \langle \partial_k \psi | \partial_l \psi \rangle
+            - \langle\partial_k \psi | \psi \rangle \langle\psi | \partial_l \psi \rangle.
```

Comment:
This is a minor comment, but I think `l` always looks weird in latex. We could e.g. either use `\ell` which looks nicer or just use indices `jk` instead of `kl`, if we want 🙂

Reply:
I copy-and-pasted from the old code. I think `ij` seems better (I don't know why `ij` is not used).